tests/kafkatest/services/kafka/kafka.py

```diff
@@ -91,12 +116,11 @@ def stop_node(self, node, clean_shutdown=True):
 
         for pid in pids:
             node.account.signal(pid, sig, allow_fail=False)
-
-        node.account.ssh("rm -f /mnt/kafka.pid", allow_fail=False)
+        wait_until(lambda: len(self.pids(node)) == 0, timeout_sec=20, err_msg="Kafka node failed to stop")
 
     def clean_node(self, node):
         node.account.kill_process("kafka", clean_shutdown=False, allow_fail=True)
-        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log /mnt/kafka.pid", allow_fail=False)
+        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log", allow_fail=False)
```

Comment:
What I really want is for the cleanup command to not have a bunch of hard-coded filenames since it's very easy for cleanup to get out of sync with the files that are actually created.

As for your jmx mixin example - during normal operation, the I think jmx processes should be stopped before any cleanup code is invoked, so I don't believe the order of data removal matters?